ls1intum/Artemis
# Feature/tutor assessment tracking

### Checklist


### Motivation and Context


Currently, changes of automatic feedback in text-exercises is not tracked. This PR introduces the connection to an external tracking component placed in Athene. There the data is collected and will be analysed at a later point. This way there are no modifications to the Artemis database scheme and it has no impact on the performance of the Artemis server.

### Description

#### Server
- Add profile "athene"
- Create a jwt token and add an extra header to the assessment route when profile "athene" is set

#### Client
- Store the jwt token in the participation
- Send a tracking request (Fire & Forget) with the result when the new header is set and the feedback is loaded, saved or submitted
- Route the tracking request directly to athene using nginx **(NOTE: This is done on the server, you cannot see it in the code!)**


### Steps for Testing on Testserver 1 or 3

Start Artemis locally with "ARTEMIS (SERVER & TEXT CLUSTERING)
1. Log in to Artemis
2. Navigate to Course Administration
3. Assess a submission for a text exercise
4. See that tracking request is send in your browser dev tools (There will be an error because nginx is not running locally)
5. See that you don't have a console error

### Nginx Config:
```
location /athene-tracking {
        proxy_pass https://athene01.ase.in.tum.de/tracking;
        proxy_set_header Host $host;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto $scheme;
    }
```
